core-bundle/src/Image/Studio/Figure.php

```diff
@@ -0,0 +1,304 @@
+                'addImage' => true,
+            ]
+        );
+
+        // Context sensitive properties
+        if (null !== ($title = StringUtil::specialchars($metaData->getTitle()))) {
+            $templateData['picture']['title'] = $title;
+        }
+
+        if (null !== $floatingProperty) {
+            $templateData['floatClass'] = " float_{$floatingProperty}";
+        }
+
+        // Link attributes
+        if (!empty($linkAttributes)) {
+            $htmlAttributes = array_map(
+                static function (string $attribute, string $value) {
+                    return sprintf('%s="%s"', $attribute, $value);
+                },
+                array_keys($linkAttributes), $linkAttributes
+            );
+
+            $templateData['attributes'] = ' '.implode(' ', $htmlAttributes);
+        }
+
+        if ('' !== ($href = $this->getLinkHref())) {
+            $templateData['href'] = $href;
+        }
+
+        // Light box
+        if ($this->hasLightBox()) {
+            if (!empty($templateData['imageTitle']) && empty($templateData['linkTitle'])) {
+                $templateData['linkTitle'] = $templateData['imageTitle'];
+                unset($templateData['imageTitle']);
+            }
+
+            $lightBox = $this->getLightBox();
+
+            if ($lightBox->hasImage()) {
+                $image = $lightBox->getImage();
+
+                $templateData['lightboxPicture'] = [
+                    'img' => $image->getImg(),
+                    'sources' => $image->getSources(),
+                ];
+            }
+        }
+
+        return $templateData;
+    }
+
+    /**
+     * Apply the legacy template data to an existing Contao template. This
+     * handles special cases to prevent overriding existing keys / values.
+     *
+     * Note: Do not use this method when using modern/Twig templates! Instead,
+     *       add this object to your template's context and directly access the
+     *       specific data you need.
+     */
+    public function applyLegacyTemplateData(Template $template, $floatingProperty = null, $marginProperty = null): void
+    {
+        $new = $this->getLegacyTemplateData($floatingProperty, $marginProperty);
+        $existing = $template->getData();
+
+        // Do not override the "href" key (see #6468)
+        if (isset($new['href'], $existing['href'])) {
+            $new['imageHref'] = $new['href'];
+            unset($new['href']);
+        }
+
+        // Append attributes instead of replacing
+        if (isset($new['attributes'], $existing['attributes'])) {
+            $new['attributes'] = ($existing['attributes'] ?? '').$new['attributes'];
+        }
+
+        $template->setData(array_replace_recursive($existing, $new));
+    }
+
+    /**
+     * Evaluate Closure to retrieve the value.
+     */
+    private function resolveIfClosure(&$property): void
+    {
+        if ($property instanceof Closure) {
+            $property = $property($this);
```

Comment:
Is there a specific reason why the callback should need the figure instance as a parameter?
```suggestion
            $property = $property();
```

Reply:
Yes, that's because you might want to depend your values on other properties. I'm doing this in the `FigureBuilder for example with the meta data: https://github.com/contao/contao/blob/f1bda9392947d5e9bbea0a2dbf1f2d65a176064d/core-bundle/src/Image/Studio/FigureBuilder.php#L391